pf/internal/schemashim/object_pseudoresource.go

```diff
@@ -129,15 +126,16 @@ func (r *objectPseudoResource) GetOk(key string) (shim.Schema, bool) {
 		return newBlockSchema(key, block), true
 	}
 
-	// If there fail to find an Attr, perhaps we have a simple ObjectType then we can look up the property's
-	// AttrType and go with that.
-	if objType, ok := r.obj.(types.ObjectType); ok {
-		if t, ok := objType.AttrTypes[key]; ok {
-			return newTypeSchema(t, nil), true
-		}
+	if t, err := r.obj.ApplyTerraform5AttributePathStep(tftypes.AttributeName(key)); err == nil {
+		typ, ok := t.(attr.Type)
```

Comment:
As I understand it `r.obj.ApplyTerraform5AttributePathStep(tftypes.AttributeName(key))` represents a `.key` access in the same space value space (`attr.Type`) as `r.obj`.

By working within `attr.Type` (instead of down-casting to `types.ObjectType`, we get support for custom implementations of `basetypes.ObjectTypeable`.

Reply:
Yeah this seems right.